containers/Dockerfile.buildenv

```diff
@@ -37,6 +37,14 @@ RUN apt-get update -y && apt-get install -y libc++-dev libc++1 libc++abi-dev \
 RUN apt-get install -y ccache doxygen python-pip && pip install coverxygen \
     && pip install numpy
 
+# Install bibtex for Doxygen bibliography management
+# [TinyTex](https://github.com/yihui/tinytex) provides a lean TeXLive profile,
```

Comment:
I think we should stick to one of the official TeX distributions, not something somebody hacked together. What's wrong with a minimal TexLive installation?

Reply:
The only thing wrong about it is that it’s at least ~500MB, if that’s OK we can just do `apt-get install biber` ;)

This script actually installs TeXLive with a custom configuration that strips it down to a minimum working version.

Reply:
Are you sure you cannot get something smaller without relying on random repos?

Reply:
I looked pretty extensively but may have missed something of course.

What we can do is replicate what that repo does ourselves, which is just downloading TeXLive, configuring it with a profile file, let it perform the installation and then installing a bunch of tex packages. It’s copy-pasting from that repo.

Reply:
I'd be happier having the shell script in our repo so we don't have another dependency. The script should contain a link to the repo and the appropriate license and copyright info.

Reply:
These sizes all sound huge to me.  My distro's package containing BibTeX is 62 MB (including a basic TeX Live installation), with probably a few more MB for dependencies.  Possibly it's hard to get down to that level without putting a fair amount of effort into packaging.

Reply:
Okay, that's much more in line with what I was expecting. @wthrowe we could also install the minimal tex during a travis build for the documentation since that one goes fairly quick so that the image doesn't get larger. However, we then run the risk of having our tests fail if the place where we are installing from goes down. What are your thoughts on that?

Reply:
Our containers are based on Ubuntu, so does Ubuntu provide a reasonable minimal TeX installation?  We can probably count on Ubuntu not disappearing on us.